### PR TITLE
Add experimental 3D Tiles Next features to the documentation and changelog

### PR DESCRIPTION
**Pull request by ptrgags**
As part of [3D Tiles Next](https://github.com/CesiumGS/3d-tiles/tree/main/next) efforts, we want to document the parts of these new features that can be interacted with today. This PR exposes some of the public (but experimental) API so it appears in the documentation.

@sanjeetsuhag could you review?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+
+- Added experimental class `ModelExperimental`, a new architecture for `Model` to support new glTF extensions.
+- Added experimental implementation of [3D Tiles Next extensions](https://github.com/CesiumGS/3d-tiles/tree/main/next) for 3D Tiles and glTF:
+  - [3DTILES_metadata](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_metadata) for adding structured metadata to tilesets, tiles, or groups of tile content
+  - [3DTILES_multiple_contents](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_multiple_contents) for storing multiple contents within a single tile
+  - (glTF extension) [EXT_mesh_features](https://github.com/KhronosGroup/glTF/pull/2082) for adding semantic metadata to glTF models and tile contents. This is the glTF equivalent of `3DTILES_metadata`. Requires use of `ModelExperimental`.
+  - [3DTILES_implicit_tiling](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_implicit_tiling) for a compact representation of quadtrees and octrees
+  - [3DTILES_bounding_volume_S2](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_bounding_volume_S2) for [S2](https://s2geometry.io/) bounding volumes. These
```
Comment (sanjeetsuhag) [inline CHANGES.md:13]:
```suggestion
  - [3DTILES_bounding_volume_S2](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_bounding_volume_S2) for [S2](https://s2geometry.io/) bounding volumes.
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
-### 1.87 - 2021-11-1
+### 1.87.1 - 2021-11-09
+
+##### Additions :tada:
+
+- Added experimental class `ModelExperimental`, a new architecture for `Model` to support new glTF extensions.
+- Added experimental implementation of [3D Tiles Next extensions](https://github.com/CesiumGS/3d-tiles/tree/main/next) for 3D Tiles and glTF:
+  - [3DTILES_metadata](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_metadata) for adding structured metadata to tilesets, tiles, or groups of tile content
+  - [3DTILES_multiple_contents](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_multiple_contents) for storing multiple contents within a single tile
+  - (glTF extension) [EXT_mesh_features](https://github.com/KhronosGroup/glTF/pull/2082) for adding semantic metadata to glTF models and tile contents. This is the glTF equivalent of `3DTILES_metadata`. Requires use of `ModelExperimental`.
```
Comment (sanjeetsuhag) [inline CHANGES.md:11]:
```suggestion
  - [EXT_mesh_features](https://github.com/KhronosGroup/glTF/pull/2082) (glTF extension)  for adding semantic metadata to glTF models and tile contents. This is the glTF equivalent of `3DTILES_metadata`. Requires use of `ModelExperimental`.
```

| Before | After |
| - | - |
| image | image |

I just think this looks slightly better.

File: Source/Core/ExperimentalFeatures.js [hunk 3]
```diff
@@ -12,15 +12,12 @@
  *   <li>Experimental flags should be short-lived. Make it clear in the PR what it would take to promote the feature to a regular feature.</li>
  *   <li>To avoid cluttering the code, check the flag in as few places as possible. Ideally this would be a single place.</li>
  * </ul>
- *
- * @private
  */
```
Comment (sanjeetsuhag) [inline Source/Core/ExperimentalFeatures.js:15]: I think we need an @experimental tag here.

image

File: Source/Scene/Cesium3DTileFeature.js [hunk 4]
```diff
@@ -212,7 +212,30 @@ Cesium3DTileFeature.prototype.getProperty = function (name) {
...
+ *   <li>Group metadata property by semantic</li>
+ *   <li>Group metadata property by property ID</li>
+ *   <li>Tileset metadata property by semantic</li>
+ *   <li>Tileset metadata property by property ID</li>
+ *   <li>Otherwise, return undefined</li>
+ * </ol>
+ *
+ * @param {Cesium3DTileContent} content The content for accessing the metadata
+ * @param {Number} batchId The batch ID (or feature ID) of the feature to get a property for
+ * @param {String} name The semantic or property ID of the feature. Semantics are checked before property IDs in each granularity of metadata.
```
Comment (sanjeetsuhag) [inline Source/Scene/Cesium3DTileFeature.js:235]: The parameter is `name` but the description calls it `id`. We should pick one.

File: Source/Scene/Cesium3DTileset.js [hunk 5]
```diff
@@ -1316,17 +1316,18 @@ Object.defineProperties(Cesium3DTileset.prototype, {
   },
 
   /**
-   * A custom shader to apply to the tileset. Only used for contents that use
-   * {@link ModelExperimental}. Using custom shaders with a {@link Cesium3DTileStyle}
-   * may lead to undefined behavior.
+   * A custom shader to apply to all tiles in the tileset. Only used for
```
Comment (sanjeetsuhag) [inline Source/Scene/Cesium3DTileset.js:1319]: Maybe add a reference to `ExperimentalFeatures` here, so people know to enable it.

File: Source/Scene/ModelExperimental/ModelFeature.js [hunk 6]
```diff
@@ -147,7 +149,20 @@ ModelFeature.prototype.getProperty = function (name) {
 };
 
 /**
- * @private
+ * Returns a copy of the feature's property with the given name, examining all
+ * the metadata from the EXT_mesh_features and legacy EXT_feature_metadata glTF
```
Comment (sanjeetsuhag) [inline Source/Scene/ModelExperimental/ModelFeature.js:153]: We should add a link to the extensions here.

File: Source/Scene/Cesium3DTileset.js [hunk 7]
```diff
@@ -1316,17 +1316,18 @@ Object.defineProperties(Cesium3DTileset.prototype, {
...
+   * contents that use {@link ModelExperimental}. Using custom shaders with a
+   * {@link Cesium3DTileStyle} may lead to undefined behavior. Any primitive
+   * that is incompatible with the custom shader code (e.g. due to missing
+   * vertex attributes) will be rendered as if no custom shader were applied.
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js:1323]: > Any primitive that is incompatible with the custom shader code (e.g. due to missing vertex attributes) will be rendered as if no custom shader were applied

Is this true? I thought default attribute values were used.

File: Source/Scene/ModelExperimental/ModelExperimental.js [hunk 8]
```diff
@@ -311,8 +309,6 @@ Object.defineProperties(ModelExperimental.prototype, {
    *
    * @type {Cesium3DTileContent}
    * @readonly
-   *
-   * @private
    */
   content: {
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelExperimental.js:313]: `content` and `featureTableId` should stay private

File: Source/Scene/Cesium3DTileFeature.js (hunk 4 above)
Comment (ptrgags) [reply Source/Scene/Cesium3DTileFeature.js:235]: We talked about this in Slack, this change would be a larger one, really more in the scope of #9914.

File: Source/Scene/ModelExperimental/ModelFeature.js [hunk 9]
```diff
@@ -10,7 +10,7 @@ import defined from "../../Core/defined.js";
  * Modifications to a <code>ModelFeature</code> object have the lifetime of the model.
  * </p>
  * <p>
- * Do not construct this directly. Access it through {@link ModelFeatureTable#getFeature}, {@link Cesium3DTileContent#getFeature} or
+ * Do not construct this directly. Access it through {@link Cesium3DTileContent#getFeature} or
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelFeature.js:13]: > Access it through {@link Cesium3DTileContent#getFeature}

This part may not be relevant anymore. `Cesium3DTileContent#getFeature` will return a `Cesium3DTileFeature` not a `ModelFeature`. @sanjeetsuhag can you confirm?

File: Source/Scene/ModelExperimental/ModelFeature.js [hunk 10]
```diff
@@ -147,7 +149,20 @@ ModelFeature.prototype.getProperty = function (name) {
...
+ *
+ * <ol>
+ *   <li>Feature metadata property by semantic</li>
+ *   <li>Feature metadata property by property ID</li>
+ * </ol>
+ *
+ * @param {String} name The semantic or property ID of the feature. Semantics are checked before property IDs in each granularity of metadata.
+ * @return {*} The value of the property or <code>undefined</code> if the feature does not have this property.
+ *
+ * @experimental This feature is using part of the 3D Tiles spec that is not final and is subject to change without Cesium's standard deprecation policy.
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelFeature.js:165]: We may want to rename `name` to `id`

File: CHANGES.md [hunk 11]
```diff
@@ -1,6 +1,20 @@
...
+  - [3DTILES_implicit_tiling](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_implicit_tiling) for a compact representation of quadtrees and octrees
+  - [3DTILES_bounding_volume_S2](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_bounding_volume_S2) for [S2](https://s2geometry.io/) bounding volumes. These
+  - [3DTILES_content_gltf](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_content_gltf) for using glTF models directly as tile contents
+- Added experimental `CustomShader` class for styling `Cesium3DTileset` or `ModelExperimental` with custom GLSL shaders.
```
Comment (lilleyse) [inline CHANGES.md:15]: I did a pass over the changes to help simplify a bit. How does this sound?

> - Added experimental implementation sof [3D Tiles Next](https://github.com/CesiumGS/3d-tiles/tree/main/next). The following extension are supported:
>   - [3DTILES_content_gltf](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_content_gltf) for using glTF models directly as tile contents
>   - [3DTILES_metadata](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_metadata) for adding structured metadata to tilesets, tiles, or groups of tile content
>   - [EXT_mesh_features](https://github.com/KhronosGroup/glTF/pull/2082) for adding feature identification and feature metadata to glTF models
>   - [3DTILES_implicit_tiling](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_implicit_tiling) for a compact representation of quadtrees and octrees
>   - [3DTILES_bounding_volume_S2](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_bounding_volume_S2) for [S2](https://s2geometry.io/) bounding volumes
>   - [3DTILES_multiple_contents](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_multiple_contents) for storing multiple contents within a single tile
> - Added `ModelExperimental`, a new experimental architecture for loading glTF models
> - Added `CustomShader` class for styling `Cesium3DTileset` or `ModelExperimental` with custom GLSL shaders
> - Added `ExperimentalFeatures.enableModelExperimental` for enabling certain experimental features. `ModelExperimental`, `EXT_mesh_features`, and custom shaders are only enabled when this flag is true.

File: Source/Scene/ModelExperimental/ModelFeature.js (hunk 10 above)
Comment (ptrgags) [reply Source/Scene/ModelExperimental/ModelFeature.js:165]: Probably, but talking to @sanjeetsuhag about a similar thing in Cesium3DTileFeature, it's probably best handled with #9914

File: Source/Scene/ModelExperimental/ModelFeature.js (hunk 9 above)
Comment (sanjeetsuhag) [reply Source/Scene/ModelExperimental/ModelFeature.js:13]: Yes, that is correct. If part of 3D Tiles, `Cesium3DTileFeatures` will be returned. [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bVJNj9MwEP0rQ06ptHKEuNFsBWS7YkURh0bsJRJy7cnW4NiRPcnSRf3v2PlATdlDYs/Hm5n3PAV61TVs+7tFpxo0xPU9cuoceoaGHzR+tRL1ZRxugVyH68oIazxBr/AZXfAafIZirPd98KVVIga7sAGoDLoqWQXcjCSl0SMtoePx7q4cg+mfygB0Tr+HKsn2LQqf3XHi2SIv+8RJHFHO56Oio+1osMrIIpt6sZ/emiqpzHkcZByeeYEGWRsoKlJ94M6lTCdITJzSXqxtSnsZCCHuwKNGQSgn6ZaVHaLZt1zgtg/6feZG6uhHejBtRx8FKWvSujPDBazZYU2FVuJXGpk3tseoe2VWMEgR+7Uh+q9bkG/JIkJnGGutV7FwHBZA1ZC+mYSWWIcnkemi2GrqAuAw2GZAneMvflc8Q+cFeD3nxee1Gpm2T+kVJs5xvpkfe3+lTnlqke229+WPYvdQfFmtk5sk93TSuBmnAvigmtY6iiuRMpYRNq3mFFbg0IVRiAnvR64AeXYJzaXqQcnbV5YShObeh0jdab1XL1glmzwL+f9BteVSmadvPTrNTzHt+HazG52MsTwL5utIslYfuLuq/Bc)

File: Source/Scene/Cesium3DTileset.js (hunk 7 above)
Comment (ptrgags) [reply Source/Scene/Cesium3DTileset.js:1323]: Ah right the default values... probably best if we omit the implementation details, it's a bit complex

**Pull request merged**